baseplate/lib/secrets.py

```diff
@@ -339,21 +393,22 @@ def make_object_for_context(self, name: str, span: Span) -> "SecretsStore":
            baseplate.add_to_context("secrets", secrets)
 
         """
-        return _CachingSecretsStore(self._filewatcher)
+        return _CachingSecretsStore(self._filewatchers)
 
 
 class _CachingSecretsStore(SecretsStore):
     """Lazily load and cache the parsed data until the server span ends."""
 
-    def __init__(self, filewatcher: FileWatcher):  # pylint: disable=super-init-not-called
-        self._filewatcher = filewatcher
+    def __init__(
+        self, filewatchers: Dict[str, FileWatcher]
+    ):  # pylint: disable=super-init-not-called
+        self._filewatchers = filewatchers
+        self._cached_data: Dict[str, Tuple[Dict, float]] = {}
 
-    @cached_property
-    def _data(self) -> Tuple[Any, float]:
-        return super()._get_data()
-
-    def _get_data(self) -> Tuple[Dict, float]:
-        return self._data
+    def _get_data(self, filename: str) -> Tuple[Dict, float]:
+        if not self._cached_data[filename]:
+            self._cached_data[filename] = super()._get_data(filename)
+        return self._cached_data[filename]
```

Comment:
won't this raise a KeyError if the data's not cached? 

something like this would be more efficient in lookups since we don't hit the dictionary multiple times:

```suggestion
        try:
            result = self._cached_data[filename]
        except KeyError:
            result = super()._get_data(filename)
            self._cached_data[filename] = result
        return result
```

(though note this is still slower than `@cached_property` for the single file case)

Reply:
decided to use `.get()` rather than the try/except, I believe that's slightly more performant.